1341-MoveAlmaTransformationToFix

TobiasNx opened the pull request:
Transform the describedBy morph into fix. Also move all other transformation to fix.

See #1341.

TobiasNx commented on src/main/resources/alma/fix/subjects.fix:
```diff
@@ -0,0 +1,397 @@
+end
+
+# TODO: Find out why this does not work?
+# lookup("subject[].*.componentList[].*.type[].*","rswk-indicator")
+
```
@blackwinter do you have an idea why this breaks?

TobiasNx replied:
```
Caused by: java.lang.IllegalArgumentException: Using ref, but can't find: 2 in: null
        at org.metafacture.metafix.FixPath.insertInto(FixPath.java:298)
        at org.metafacture.metafix.FixPath.insertInto(FixPath.java:265)
        at org.metafacture.metafix.FixPath.lambda$insertInto$20(FixPath.java:293)
        at org.metafacture.metafix.Value$TypeMatcher.match(Value.java:344)
        at org.metafacture.metafix.Value$TypeMatcher.ifArray(Value.java:317)
        at org.metafacture.metafix.FixPath.lambda$insertInto$22(FixPath.java:293)
```

TobiasNx replied:
It runs with: `$ mvn clean install -DskipTests=true ; mvn -DgenerateTestData=true failsafe:integration-test -Dit.test=AlmaMarc21XmlToLobidJsonMetafixTest.java`

fsteeg replied:
I'm afraid I have no quick fix here, it's actually surprising how that current path update logic works at all, I'll keep trying.

TobiasNx replied:
Thanks this seems to work now!

TobiasNx commented:
Okay, this seems like a very huge branch and MR.
But besides the rest of TODOs the main construction for the new transformation seems done.

While changing the ALMA-transformation to fix I stumbled over some problems for that I needed to go back to the old ALEPH Morph especially with the `medium` and `type` elements. Here are some questions open concerning the mappings and the possibilities it seems that some info from Aleph is lost "forever". Others seem to have no proper mappings.

Things I suggest to do now is create an index from the new fix transformation. Repair all broken things with real data. Then start finishing and cleaning.

TobiasNx commented:
Added `hasItem` POR transformation also changed `oclcNumber` to array instead of `string`

Note to @hagbeck : when we move the ALMA Transformation to Fix, oclcNumber will be an array. Should we already implement this in the old ALMA-Morph, or is this change not relevant to you?

hagbeck commented:
@TobiasNx We are using the `oclcNumber` but the changes are not relevant for us now. So no changing of the ALMA-Morph is necessary.

dr0i commented:
Merged into master. We will go on using smaller PRs to solve expected issues. Closed.
